crates/polars-core/src/series/any_value.rs

```diff
@@ -493,9 +493,18 @@ fn any_values_to_list(
     let mut valid = true;
     #[allow(unused_mut)]
     let mut out: ListChunked = if inner_type == &DataType::Null {
+        let mut first_value_dtype = DataType::Null;
         avs.iter()
             .map(|av| match av {
-                AnyValue::List(b) => Some(b.clone()),
+                AnyValue::List(b) => {
+                    if !b.is_empty() && matches!(first_value_dtype, DataType::Null) {
+                        first_value_dtype = b.dtype().clone();
+                    }
+                    if b.is_empty() || b.null_count() == b.len() {
```

Comment:
This can be an `else if` right?

Why do we need to cast if the null count is equal to len? I think it should be based on types, not values?

Reply:
The problematic cases are not only like this:
`pl.Series("a", [["a"], []], dtype=pl.List)`
but also cases like this:
`pl.Series("a", [["a"], [None, None]], dtype=pl.List)`
The null count check addresses the second case.
About the `else if`, yes, it should work equally well.